modules/data.atmosphere/R/read.register.R

```diff
@@ -35,10 +35,11 @@ read.register <- function(register.xml, con) {
     } else if ((!is.null(register$format$id) & is.null(register$format$name)) 
                | 
                (!is.null(register$format$id) & is.null(register$format$mimetype))) {
-      register$format$name <- PEcAn.DB::db.query(
-        paste("SELECT name from formats where id = ", register$format$id), con)[[1]]
-      register$format$mimetype <- PEcAn.DB::db.query(
-        paste("SELECT mime_type from formats where id = ", register$format$id), con)[[1]]
+      # Retrieve format name and mimetype from the database
+      register$format <- PEcAn.DB::db.query(
+        paste("SELECT name, mime_type AS mimetype FROM formats WHERE id = ", register$format$id), con
+      )
```

Comment:
Two problems here: 
1. Assigning the result to register$format will overwrite the existing format object (which contains at least `id` and `inputtype` -- see lines 25 and 35) with one that contains only `name` and `mimetype`. Better to query once into a temporary result and then assign from there into `register$format$name` and `register$format$mimetype.
2. The existing line 41 is broken because the formats table doesn't have a `mime_type` column, it has a `mimetype_id` tht then needs to be looked up in the `mimetypes` table. I think you want `"SELECT name, type_string AS mimetype FROM formats JOIN mimetypes ON formats.mimetype_id = mimetypes.id  WHERE formats.id ="`

Reply:
1. I seemed to miss it. Fixing it now.
2. Ahh, I recall us having a conversation about this earlier and that seems to be the case.